### PR TITLE
II 1993 replace .bashrc with config.ini

### PR DESCRIPTION
**Pull request by yqian10**

### CODE REVIEW

File: elasticsearch-log/config.ini [hunk 1]
```diff
@@ -0,0 +1,6 @@
+#example for config
+[elasticsearch-log]
+insightFinder_license_key = 10261993s
+insightFinder_project_name = elastic
+insightFinder_user_name = yqian10
+sampling_interval = 10
```
Comment (zhangzinan) [inline elasticsearch-log/config.ini]: add new line at end.

File: elasticsearch-log/get_logs_elasticsearch.py [hunk 2]
```diff
@@ -227,35 +228,32 @@ def filter(self, record):
...
-        samplingIntervalLine = fileContent[4].split(" ")
-        if len(samplingIntervalLine) != 2:
-            logger.error("Agent not correctly configured(sampling interval). Check .agent.bashrc file.")
-            sys.exit(1)
-        configVars['samplingInterval'] = samplingIntervalLine[1].split("=")[1].strip()
+    try:
+        if os.path.exists(os.path.join(homepath, "elasticsearch-log", "config.ini")):
+            parser = SafeConfigParser()
+            parser.read(os.path.join(homepath, "elasticsearch-log", "config.ini"))
+            insightFinder_license_key = parser.get('elasticsearch-log', 'insightFinder_license_key')
```
Comment (zhangzinan) [inline elasticsearch-log/get_logs_elasticsearch.py]: if use Camel-Case, do not use underscore again.

File: elasticsearch-log/config.ini [hunk 3]
```diff
@@ -1,6 +1,6 @@
...
 [elasticsearch-log]
-insightFinder_license_key = 10261993s
-insightFinder_project_name = elastic
-insightFinder_user_name = yqian10
-sampling_interval = 10
\ No newline at end of file
+insightFinder_license_key =
+insightFinder_project_name =
+insightFinder_user_name =
+sampling_interval =
```
Comment (zhangzinan) [inline elasticsearch-log/config.ini]: add newline end

File: common/config.ini [hunk 4]
```diff
@@ -0,0 +1,6 @@
+#sample config
+[metrics]
+insightFinder_license_key =
+insightFinder_project_name =
+insightFinder_user_name =
+sampling_interval =
```
Comment (zhangzinan) [inline common/config.ini]: add newline end

Comment (zhangzinan) [inline common/config.ini]: add newline end

File: elasticsearch-log/get_logs_elasticsearch.py [hunk 5]
```diff
@@ -227,35 +228,32 @@ def filter(self, record):
...
-            logger.error("Agent not correctly configured(username). Check .agent.bashrc file.")
-            sys.exit(1)
-        configVars['userName'] = userNameLine[1].split("=")[1].strip()
-        # get sampling interval
-        samplingIntervalLine = fileContent[4].split(" ")
-        if len(samplingIntervalLine) != 2:
-            logger.error("Agent not correctly configured(sampling interval). Check .agent.bashrc file.")
-            sys.exit(1)
-        configVars['samplingInterval'] = samplingIntervalLine[1].split("=")[1].strip()
+    try:
```
Comment (zhangzinan) [inline elasticsearch-log/get_logs_elasticsearch.py]: f use Camel-Case, do not use underscore again.

**Comment by ptmr3** [comment]
@zhangzinan @moharnab123saikia please review and approve when everything is ready for my review and merge, thank you

File: cadvisor/getmetrics_cadvisor.py [hunk 6]
```diff
@@ -24,12 +25,15 @@
...
-    os.environ[key] = value.strip()
-proc.communicate()
+try:
+    if os.path.exists(os.path.join(homepath, "cadvisor", "config.ini")):
+        parser = SafeConfigParser()
+        parser.read(os.path.join(homepath, "cadvisor", "config.ini"))
+        sampling_interval = parser.get('cadvisor', 'sampling_interval')
+        os.environ["SAMPLING_INTERVAL"] = sampling_interval
+except IOError:
+        print("config.ini file is missing")
```
Comment (moharnab123saikia) [inline cadvisor/getmetrics_cadvisor.py]: Use logger from other code.

File: cadvisor/getmetrics_cadvisor.py [hunk 7]
```diff
@@ -24,12 +25,15 @@
...
-for line in proc.stdout:
-    (key, _, value) = line.partition("=")
-    os.environ[key] = value.strip()
-proc.communicate()
+try:
+    if os.path.exists(os.path.join(homepath, "cadvisor", "config.ini")):
+        parser = SafeConfigParser()
+        parser.read(os.path.join(homepath, "cadvisor", "config.ini"))
+        sampling_interval = parser.get('cadvisor', 'sampling_interval')
+        os.environ["SAMPLING_INTERVAL"] = sampling_interval
```
Comment (moharnab123saikia) [inline cadvisor/getmetrics_cadvisor.py]: environment variable setting can be ommitted/may be not needed. Refer other newly refactored agents agents

File: cadvisor/getmetrics_cadvisor.py [hunk 8]
```diff
@@ -24,12 +25,15 @@
...
+except IOError:
+        print("config.ini file is missing")
+
 samplingInterval = os.environ["SAMPLING_INTERVAL"]
```
Comment (moharnab123saikia) [inline cadvisor/getmetrics_cadvisor.py]: naming in underscore.
Sampling interval can be directly taken from config file

File: collectd/collectdReportMetrics.py [hunk 9]
```diff
@@ -36,12 +37,28 @@
...
+        if len(insightFinder_license_key) == 0:
+            print("Agent not correctly configured(license key). Check config file.")
+            sys.exit(1)
+        if len(insightFinder_project_name) == 0:
+            print("Agent not correctly configured(project name). Check config file.")
+            sys.exit(1)
+        if len(insightFinder_user_name) == 0:
+            print("Agent not correctly configured(username). Check config file.")
+            sys.exit(1)
+        os.environ["INSIGHTFINDER_LICENSE_KEY"] = insightFinder_license_key
```
Comment (moharnab123saikia) [inline collectd/collectdReportMetrics.py]: Environment variables need to be removed and replaced from the config file

File: collectd/collectdReportMetrics.py [hunk 10]
```diff
@@ -36,12 +37,28 @@
...
+        os.environ["INSIGHTFINDER_PROJECT_NAME"] = insightFinder_project_name
+        os.environ["INSIGHTFINDER_USER_NAME"] = insightFinder_user_name
+except IOError:
+        print("config.ini file is missing")
 
 LICENSEKEY = os.environ["INSIGHTFINDER_LICENSE_KEY"]
```
Comment (moharnab123saikia) [inline collectd/collectdReportMetrics.py]: Remove environment variables. Dont use All caps as it is not constant

File: collectd/collectdReportMetrics.py [hunk 11]
```diff
@@ -394,3 +411,5 @@ def isStrInKeys(mydict, mystr):
     print "Custom metrics sent"
```
Comment (moharnab123saikia) [inline collectd/collectdReportMetrics.py]: Use logging class

File: common/reportMetrics.py [hunk 12]
```diff
@@ -101,35 +102,32 @@ def getParameters():
 
 def getAgentConfigVars():
```
Comment (moharnab123saikia) [inline common/reportMetrics.py]: naming format

File: elasticsearch-log/get_logs_elasticsearch.py (hunk 5 above)
Comment (moharnab123saikia) [reply elasticsearch-log/get_logs_elasticsearch.py]: @zhangzinan We have decided to convert all naming to pep8 standard format which is underscore

File: kafka/getmetrics_kafka.py [hunk 13]
```diff
@@ -81,13 +81,13 @@ def get_agent_config_vars(normalization_ids_map):
         if os.path.exists(os.path.join(parameters['homepath'], "kafka", "config.ini")):
             parser = SafeConfigParser()
             parser.read(os.path.join(parameters['homepath'], "kafka", "config.ini"))
-            insightFinder_license_key = parser.get('kafka', 'insightFinder_license_key')
-            insightFinder_project_name = parser.get('kafka', 'insightFinder_project_name')
-            insightFinder_user_name = parser.get('kafka', 'insightFinder_user_name')
-            sampling_interval = parser.get('kafka', 'sampling_interval')
-            group_id = parser.get('kafka', 'group_id')
+            licenseKey = parser.get('kafka', 'insightFinder_license_key')
```
Comment (moharnab123saikia) [inline kafka/getmetrics_kafka.py]: revert back to underscore

File: kafka_logs/getlogs_kafka.py [hunk 14]
```diff
@@ -84,12 +84,12 @@ def get_agent_config_vars():
         if os.path.exists(os.path.join(parameters['homepath'], "kafka_logs", "config.ini")):
             parser = SafeConfigParser()
             parser.read(os.path.join(parameters['homepath'], "kafka_logs", "config.ini"))
-            insightFinder_license_key = parser.get('kafka', 'insightFinder_license_key')
-            insightFinder_project_name = parser.get('kafka', 'insightFinder_project_name')
-            insightFinder_user_name = parser.get('kafka', 'insightFinder_user_name')
-            sampling_interval = parser.get('kafka', 'sampling_interval')
-            client_id = parser.get('kafka', 'client_id')
-            group_id = parser.get('kafka', 'group_id')
+            licenseKey = parser.get('kafka', 'insightFinder_license_key')
```
Comment (moharnab123saikia) [inline kafka_logs/getlogs_kafka.py]: revert back to underscore

File: opentsdb/getmetrics_opentsdb.py [hunk 15]
```diff
@@ -42,35 +43,32 @@ def getParameters():
 
 def getAgentConfigVars():
```
Comment (moharnab123saikia) [inline opentsdb/getmetrics_opentsdb.py]: fix naming

File: opentsdb/config.ini [hunk 16]
```diff
@@ -0,0 +1,7 @@
+#example for config
+[insightFinder]
```
Comment (moharnab123saikia) [inline opentsdb/config.ini]: section name should be [insightfinder]

**Pull request merged**